bn_mp_get_i32.c

```diff
@@ -0,0 +1,15 @@
+#include "tommath_private.h"
+#ifdef BN_MP_GET_I32_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+
+/* Return the least significant 32 bits */
+int32_t mp_get_i32(const mp_int *a)
+{
+   uint64_t res = mp_get_mag32(a);
+   return (a->sign == MP_NEG) ? (int32_t)-res : (int32_t)res;
```

Comment:
Better do the (int32_t) typecast first, then the negation:  Negation of unsigned integers doesn't make much sense (I have seen compiler warnings related to this in another project, don't remember which compiler/flags)

Reply:
Remember now what warnings I saw: It was Visual Studio 2017:

bn_conversion.c
bn_conversion.c(64): warning C4146: unary minus operator applied to unsigned type, result still unsigned
bn_conversion.c(68): warning C4146: unary minus operator applied to unsigned type, result still unsigned
bn_conversion.c(72): warning C4146: unary minus operator applied to unsigned type, result still unsigned
bn_conversion.c(76): warning C4146: unary minus operator applied to unsigned type, result still unsigned

Reply:
(VS2015, too)
It also says

```text
demo/test.c(677): warning C4102: 'LBL_ERR': unreferenced label
demo/test.c(627): warning C4101: 'i': unreferenced local variable
```

Which makes no sense at all for me.

I rolled the changes back and will stash when Travis gives its OK.

Reply:
No, the cast should happen in this order. Otherwise we get undefined behavior. Furthermore negation of unsigned numbers is defined.